pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -221,7 +225,7 @@ private RelNode optimize(RelRoot relRoot, PlannerContext plannerContext) {
 
   private QueryPlan toDispatchablePlan(RelRoot relRoot, PlannerContext plannerContext, long requestId) {
     // 5. construct a dispatchable query plan.
-    StagePlanner queryStagePlanner = new StagePlanner(plannerContext, _workerManager, requestId);
+    StagePlanner queryStagePlanner = new StagePlanner(plannerContext, _workerManager, requestId, _tableCache);
     return queryStagePlanner.makePlan(relRoot);
```

Comment:
my understanding of this POC is that. the end goal is to, within StagePlanner, assign each stage a parallelism and a set of servers to run those parallelism. is this correct?